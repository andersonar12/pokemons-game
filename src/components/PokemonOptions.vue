<template>
  <section class="options-container">
    <ul>
      <li v-for="pokemon in pokemons" @click="$emit('onSelection', pokemon)" :key="pokemon.id">{{ pokemon.name }}</li>
    </ul>
  </section>
</template>

<script>
export default {
  name: "PokemonOptions",
  components: {},
  props: {
    pokemons: {
      type: Array,
      required: true,
      default:  [],
    }
  },
  data() {
    return {
    };
  },
  mounted() {},
  methods: {},
  watch: {},
};
</script>

<style scoped>
/* Pokemon Options */
ul {
  list-style-type: none;
  padding: 0;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
