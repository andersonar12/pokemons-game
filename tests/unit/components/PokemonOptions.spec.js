import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { mockPokemons } from "../mocks/pokemons.mocks.js";

describe("PokemonOptions component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: mockPokemons,
      },
    });
  });

  test("debe de hacer match con el snapshot", () => {
    console.log(wrapper.html());
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de mostrar las 5 opciones correctamente", () => {
    const liTags = wrapper.findAll("li");
    expect(liTags.length).toBe(5);

    mockPokemons.forEach((pokemon, index) => {
      expect(liTags[index].text()).toBe(pokemon.name);
    });
  });

  test("debe de emitir 'onSelection' con sus respectivos parametros al hacer click", () => {
    const [li1, li2, li3, li4, li5] = wrapper.findAll("li");

    li1.trigger("click");
    li2.trigger("click");
    li3.trigger("click");
    li4.trigger("click");
    li5.trigger("click");

    console.log(wrapper.emitted("onSelection"));
    expect(wrapper.emitted("onSelection")).toHaveLength(5);

    mockPokemons.forEach((pokemon, index) => {
      expect(wrapper.emitted("onSelection")[index]).toEqual([pokemon]);
    });
  });
});
