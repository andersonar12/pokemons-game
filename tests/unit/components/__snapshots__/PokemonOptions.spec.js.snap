// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component debe de hacer match con el snapshot 1`] = `
<section class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>pikachu</li>
    <li>venusaur</li>
    <li>charmander</li>
    <li>mew</li>
  </ul>
</section>
`;
